### PR TITLE
LLVM release_13x CI changes

### PR DESCRIPTION
**Pull request by abrahamtovarmob**
Changes to LLVM release_13x to be able to be build by github
Minor changes to code format to remove commented code and just disable it.

### CODE REVIEW

**Comment by pawosm-arm** [comment]
@bryanpkc as you're familiar with those warnings being treated as errors, can you give a hint why it is failing on X86?

**Comment by bryanpkc** [comment]
@pawosm-arm I started looking at it yesterday. The cause is that LLVM 13 shipped this patch: https://reviews.llvm.org/D100581. There are 916 such warnings in Classic Flang which need to be fixed in a subsequent PR. For now, please apply this patch to CMakeLists.txt (in this PR):
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9dc15a8..a3e03d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -284,8 +284,8 @@ if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings")
   option(WITH_WERROR "Compile with '-Werror' C compiler flag" ON)
   if (WITH_WERROR)
-     set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=misleading-indentation -Wno-error=dangling-else -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=uninitialized -Wno-error=sometimes-uninitialized -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=logical-op-parentheses -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment -Wno-error=implicit-function-declaration -Wno-error=implicit-fallthrough")
-     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=misleading-indentation -Wno-error=dangling-else -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=uninitialized -Wno-error=sometimes-uninitialized -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=logical-op-parentheses -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment -Wno-error=implicit-function-declaration -Wno-error=implicit-fallthrough")
+     set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=misleading-indentation -Wno-error=dangling-else -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=uninitialized -Wno-error=sometimes-uninitialized -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=logical-op-parentheses -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment -Wno-error=implicit-function-declaration -Wno-error=implicit-fallthrough -Wno-error=unused-but-set-variable")
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=misleading-indentation -Wno-error=dangling-else -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=uninitialized -Wno-error=sometimes-uninitialized -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=logical-op-parentheses -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment -Wno-error=implicit-function-declaration -Wno-error=implicit-fallthrough -Wno-error=unused-but-set-variable")
   endif ()
 endif ()

```

**Comment by pawosm-arm** [comment]
I don't have access to this PR, I would have to reupload this commit.

**Comment by pawosm-arm** [comment]
Are you sure this is a CMake file from this repo?!

**Comment by bryanpkc** [comment]
> Are you sure this is a CMake file from this repo?!

Yes: https://github.com/flang-compiler/flang/blob/master/CMakeLists.txt#L287-L288

**Comment by bryanpkc** [comment]
> Are you sure this is a CMake file from this repo?!

No, it's for Classic Flang. Sorry.

**Comment by bryanpkc** [comment]
I will submit a Classic Flang commit.

**Comment by pawosm-arm** [comment]
Maybe those -Werror changes just won't work with LLVM10?

**Comment by bryanpkc** [comment]
> Maybe those -Werror changes just won't work with LLVM10?

What do you mean? They have worked in the CI so far...

**Comment by pawosm-arm** [comment]
I don't know how to make it work here...

**Comment by bryanpkc** [comment]
> I don't know how to make it work here...

Merge flang-compiler/flang#1183 first, then this one will work.

**Comment by pawosm-arm** [comment]
Ah... following other threads at the same time, somehow I assumed it was already merged...

**Review by pawosm-arm — APPROVED** [review]

File: build-llvm-project.sh [hunk 1]
```diff
@@ -65,7 +66,7 @@ fi
 
 # Build and install
 mkdir -p build && cd build
-cmake $CMAKE_OPTIONS -DLLVM_ENABLE_PROJECTS="clang;openmp" ../llvm
+cmake $CMAKE_OPTIONS -DLLVM_ENABLE_PROJECTS="llvm;clang;openmp" ../llvm
```
Comment (bryanpkc) [inline build-llvm-project.sh]: I don't think this change is needed? "llvm" is not an optional project.

File: llvm/lib/CodeGen/AsmPrinter/DebugLocEntry.h [hunk 2]
```diff
@@ -214,15 +214,17 @@ class DbgValueLoc {
...
+    if (isLocation()) {
+      llvm::dbgs() << "Loc = { reg=" << Loc.getReg() << " ";
+      if (Loc.isIndirect())
+        llvm::dbgs() << "+0";
+      llvm::dbgs() << "} ";
+    } else if (isConstantInt())
+      Constant.CIP->dump();
+    else if (isConstantFP())
+      Constant.CFP->dump();
+#endif
```
Comment (bryanpkc) [inline llvm/lib/CodeGen/AsmPrinter/DebugLocEntry.h]: I think the formatting changes in DebugLocEntry.h belong in a different commit.

**Comment by pawosm-arm** [comment]
@abrahamtovarmob will you be able to follow up with introducing the post-review changes?

File: .github/workflows/pre-compile_llvm.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Pre-compile llvm
 on:
   workflow_dispatch:
   push:
-    branches: [ release_100, release_11x, release_12x ]
+    branches: [ release_100, release_11x, release_12x, release_13x ]
```
Comment (RichBarton-Arm) [inline .github/workflows/pre-compile_llvm.yml:6]: Seems like release_100 could go?

**Comment by abrahamtovarmob** [comment]
> 

Yes, it won't be a problem

**Comment by abrahamtovarmob** [comment]
> @abrahamtovarmob will you be able to follow up with introducing the post-review changes?
Yes, it won't be a problem

File: .github/workflows/pre-compile_llvm.yml (hunk 3 above)
Comment (abrahamtovarmob) [reply .github/workflows/pre-compile_llvm.yml:6]: Some time ago we were thinking to remove it, but community wanted to leave it in case someone wants to test a commit against release_100 build. However no actual test is being triggered on release_100 branch, we removed that

Comment (RichBarton-Arm) [reply .github/workflows/pre-compile_llvm.yml:6]: OK - happy to leave it in.

**Review by bryanpkc — APPROVED** [review]
LGTM.

**Comment by RichBarton-Arm** [comment]
@kiranchandramohan  - can we go ahead with this? Seems like adding CI is a no brainer and we have approvals already.

**Comment by kiranchandramohan** [comment]
I think we can merge. Adding @shivaramaarao for info.

**Comment by kiranchandramohan** [comment]
Seems to be running into an RPC error.
```
https://github.com/flang-compiler/classic-flang-llvm-project/actions/runs/1562904583
```

@abrahamtovarmob could you have a look?

**Comment by abrahamtovarmob** [comment]
> 

Sure, I will check what is going on.
